modules/renter/hostdb/dependencies.go

```diff
@@ -0,0 +1,110 @@
+package hostdb
+
+import (
+	"net"
+	"path/filepath"
+	"time"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/persist"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// These interfaces define the HostDB's dependencies. Using the smallest
+// interface possible makes it easier to mock these dependencies in testing.
+type (
+	hdbConsensusSet interface {
+		ConsensusSetSubscribe(modules.ConsensusSetSubscriber)
+	}
+	// in order to restrict the modules.TransactionBuilder interface, we must
+	// provide a shim to bridge the gap between modules.Wallet and
+	// hdbTransactionBuilder.
+	hdbWalletShim interface {
+		NextAddress() (types.UnlockConditions, error)
+		StartTransaction() modules.TransactionBuilder
+	}
```

Comment:
yes, but unfortunately Go will still complain because the type signatures do not match: http://play.golang.org/p/RKOpJ0_-9m